core/toolbox/category.js

```diff
@@ -724,14 +732,15 @@ Blockly.ToolboxCategory.prototype.getChildToolboxItems = function() {
  * @public
  */
 Blockly.ToolboxCategory.prototype.updateFlyoutContents = function(contents) {
-  if (this.hasSubcategories()) {
-    console.warn('Category can not have both flyout contents and subcategories');
-    return;
-  }
+  this.flyoutItems_ = [];
+
   if (typeof contents == 'string') {
     this.toolboxItemDef_['custom'] = contents;
   } else {
-    this.toolboxItemDef_['contents'] = Blockly.utils.toolbox.convertFlyoutDefToJsonArray(contents);
+    // Removes old custom field when contents is updated.
```

Comment:
You're trying to clear any state that's left over from the last update. Why not throw away the whole toolboxItemDef and replace it with a new one, so you're guaranteed to have clean state?

Reply:
I think there are pieces on the toolboxItemDef that we still need. For example, we need the categoryStyle on it in refreshTheme.

Reply:
Going to merge but can take another look if we decide it is possible.